### PR TITLE
Bucket limits

### PR DESCRIPTION
**Pull request by svalasovich**

### CODE REVIEW

File: packages/smart-contract/src/SmartContract.ts [hunk 1]
```diff
@@ -70,8 +73,11 @@ export class SmartContract {
         return await this.contract.api.disconnect();
     }
 
-    async bucketCreate(balance: bigint, bucketParams: string, clusterId: bigint): Promise<BucketCreatedEvent> {
-        const tx = await this.contract.tx.bucketCreate(txOptionsPay, bucketParams, clusterId);
+    async bucketCreate(balance: bigint, clusterId: bigint, bucketParams: BucketParams = new BucketParams()): Promise<BucketCreatedEvent> {
+        console.log(bucketParams)
```
Comment (upalinski) [inline packages/smart-contract/src/SmartContract.ts:77]: remove console.log?

File: packages/smart-contract/src/SmartContract.ts [hunk 2]
```diff
@@ -70,8 +73,11 @@ export class SmartContract {
...
+        bucketParams = initDefaultOptions(bucketParams);
```
Comment (upalinski) [inline packages/smart-contract/src/SmartContract.ts:78]: initDefaultBucketParams?

File: packages/smart-contract/src/options/BucketParams.ts [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+export function initDefaultOptions(bucketParams: BucketParams): BucketParams {
+    //ToDo temporary limits
+    if (bucketParams.resource > MAX_RESOURCE || bucketParams.replication > MAX_REPLICATION) {
+        throw new Error(`Exceed bucket limits: ${bucketParams}`)
+    } else if (!bucketParams.replication || !bucketParams.resource) {
+        throw new Error(`Invalid bucket params: ${bucketParams}`);
+    }
+
+    return {
+        replication: bucketParams.replication < 0 ? defaultBucketParams.replication : bucketParams.replication | 0,
```
Comment (upalinski) [inline packages/smart-contract/src/options/BucketParams.ts:24]: may be it's better to check that passed parameters are not negative?

**Review by upalinski — APPROVED** [review]

File: packages/smart-contract/src/SmartContract.ts (hunk 1 above)
Comment (svalasovich) [reply packages/smart-contract/src/SmartContract.ts:77]: removed

File: packages/smart-contract/src/SmartContract.ts (hunk 2 above)
Comment (svalasovich) [reply packages/smart-contract/src/SmartContract.ts:78]: Renamed

File: packages/smart-contract/src/options/BucketParams.ts (hunk 3 above)
Comment (svalasovich) [reply packages/smart-contract/src/options/BucketParams.ts]: updated

**Pull request merged**